### DISCUSSION TITLE
Map Config

### DISCUSSION THREAD

**Discussion by kmarsh** [maplibre/maplibre-gl-js#3820] [Q&A]
I was looking for a way to set map-level properties and reference in styles... turns out MapBox added `setConfigProperty` in v3 which looks perfect... is there something similar for MapLibre?

**Comment by HarelM** [comment]
Nope, we also can't copy code from mapbox.
Feel free to open a PR if this is something you might need.
Reading the mapbox docs, I'm not sure I fully understand what this is supposed to do...

**Reply by kmarsh** [reply]
Thanks for the reply... it allows you to set a map-level property that can be used from styles. Think a dark/light toggle that might be set on the map itself that styles could then case-match on to set lines to white or black depending on the value.

I'll investigate a PR, without copying code from Mapbox of course.

**Reply by HarelM** [reply]
Interesting idea, language change is probably one of those configurations that can be used style-wise.
We also need to define it in the spec first to let style writers know how to take it into consideration, and also somehow provide a default value.
I would recommend creating a feature suggestion in the maplibre-style-spec repo, after the style definition is finalized, we can continue with implementation here.
